ext/symengine/ruby_function.c

```diff
@@ -0,0 +1,54 @@
+#include "ruby_function.h"
+
+
+VALUE cfunction_onearg(void (*cwfunc_ptr)(basic_struct*, const basic_struct*), VALUE operand1) {
+    basic_struct *cresult;
+    VALUE result;
+
+    basic cbasic_operand1;
+    basic_new_stack(cbasic_operand1);
+    sympify(operand1, cbasic_operand1);
+
+    cresult = basic_new_heap();
+    cwfunc_ptr(cresult, cbasic_operand1);
+    result = Data_Wrap_Struct(Klass_of_Basic(cresult), NULL , cbasic_free_heap, cresult);
+    basic_free_stack(cbasic_operand1);
+
+    return result;
+}
+
+#define IMPLEMENT_ONE_ARG_FUNC(func) \
```

Comment:
Yous should undefine this at the end if you don't intend to use this macro anywhere else except this file.